File: app/src/main/java/dk/itu/moapd/virtualpiano/view/MainActivity.java
package dk.itu.moapd.virtualpiano.view;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import dk.itu.moapd.virtualpiano.R;

import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        FragmentManager fragmentManager = getSupportFragmentManager();
        Fragment fragment = fragmentManager.findFragmentById(R.id.fragment);

        if (fragment == null)
            fragmentManager.beginTransaction()
                    .add(R.id.fragment, new MainFragment())
                    .commit();
    }

}

File: app/src/main/java/dk/itu/moapd/virtualpiano/view/MainActivity.kt
package dk.itu.moapd.virtualpiano.view

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import dk.itu.moapd.virtualpiano.R

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val fragment =
            supportFragmentManager.findFragmentById(R.id.fragment)

        if (fragment == null)
            supportFragmentManager
                .beginTransaction()
                .add(
                    R.id.fragment,
                    MainFragment()
                )
                .commit()
    }

}
